### PR TITLE
feat: support updating roles via RoleServices

### LINKED ISSUE #24402: Camunda services support updating a role
### Description
Role updates should be supported in the `camunda-service` module. An example of the implementation of updating via the service module can be found [here](https://github.com/camunda/camunda/blob/main/service/src/main/java/io/camunda/service/UserServices.java#L62-L70)

### Acceptance Criteria
- [ ] A `BrokerRoleUpdateRequest` exists hint: [BrokerUserUpdateRequest](https://github.com/camunda/camunda/blob/main/zeebe/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerUserUpdateRequest.java)
  - [ ] The field(s) that can be updated for a role are [`name`]
- [ ] A method exists with signature `updateRole(final RoleDTO roleDTO)`
- [ ] The `RoleServiceTest` class should contain unit tests for the `updateRole` method - hint: an example of working with the broker commands in tests can be found [here](https://github.com/camunda/camunda/blob/main/service/src/test/java/io/camunda/service/UserServiceTest.java#L66-L81)

### PR DESCRIPTION
**Pull request by emulbreh**
## Description

The PR adds support for updating roles via `RoleServices::updateRole`.

closes #24402

### CODE REVIEW

File: service/pom.xml [hunk 1]
```diff
@@ -82,6 +82,14 @@
...
     </dependency>
 
+    <dependency>
+      <groupId>io.camunda</groupId>
+      <artifactId>zeebe-gateway</artifactId>
+      <classifier>tests</classifier>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+
```
Comment (Ben-Sheppard) [inline service/pom.xml:92]: ❓ I'm wondering what has caused this dependency addition,

Comment (emulbreh) [reply service/pom.xml:92]: That's for `StubbedBrokerClient`

Comment (Ben-Sheppard) [reply service/pom.xml:92]: Ah I see, nice addition :)

**Review by Ben-Sheppard — APPROVED** [review]
LGTM thank you for contributing so early!

**Pull request merged**